oak-search-elastic/src/test/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticConnectionRule.java

```diff
@@ -48,8 +48,16 @@ public class ElasticConnectionRule extends ExternalResource {
     private ElasticConnectionModel elasticConnectionModel;
 
     public ElasticConnectionRule(String elasticConnectionString) {
+        this(elasticConnectionString,
+                "elastic_test_" +
+                        RandomStringUtils.random(5, true, false).toLowerCase() +
+                        System.currentTimeMillis()
+        );
+    }
+
+    public ElasticConnectionRule(String elasticConnectionString, String indexPrefix) {
         this.elasticConnectionString = elasticConnectionString;
-        indexPrefix = "elastic_test_" + RandomStringUtils.random(5, true, false).toLowerCase();
+        this.indexPrefix = indexPrefix;
```

Comment:
This configuration should be used only for testing. Unfortunately, I could not find a better way to make index names consistent across executions.

In case this option is used in production, index names will clash and reindex won't work. We can potentially handle this case by detecting the fact the index name is static and instead of deleting the old one, we empty the existing and start reindexing.  This is not advisable anyway. With dynamic index names, we have the possibility to trigger reindexing and then switch the alias only when it is completed. During this time, queries will still use the old index. Static index names could lead to inconsistent results during reindexing.